neo/Consensus/ConsensusContext.cs

```diff
@@ -29,13 +29,47 @@ internal class ConsensusContext : IDisposable
         public UInt160 NextConsensus;
         public UInt256[] TransactionHashes;
         public Dictionary<UInt256, Transaction> Transactions;
-        public byte[][] Signatures;
+        public byte[][] SignedPayloads;
+        public byte[][] FinalSignatures;
         public byte[] ExpectedView;
         private KeyPair KeyPair;
         private readonly Wallet wallet;
 
         public int M => Validators.Length - (Validators.Length - 1) / 3;
 
+        /// <summary>
+        /// Local PrepareRequest original Payload
+        /// </summary>
+        public ConsensusPayload PreparePayload;
+
+        /// <summary>
+        /// Serialize PreparePayload Data into the desired PrepareRequest message
+        /// </summary>
+        public PrepareRequest GetPrepareRequestMessage(ConsensusPayload PreparePayloadToGet)
+        {
+            ConsensusMessage message;
+            try
+            {
+                message = ConsensusMessage.DeserializeFrom(PreparePayloadToGet.Data);
+            }
+            catch
+            {
+                return new PrepareRequest();
```

Comment:
Return a empty `PrepareRequest` could produce errors

Reply:
Great, @shargon! Nice catch, literally... aeuahuea

In the current implementation it is like this: https://github.com/neo-project/neo/blob/master/neo/Consensus/ConsensusService.cs#L155

What do you suggest?

Reply:
We should change tha call https://github.com/neo-project/neo/pull/426/files#diff-b09366dde7105170c26fd3757e256035R68 and do nothing with manipulated payloads

Reply:
I think that there are four places where it is being called.

Reply:
just found this TODO
```cs
            // TODO Maybe include some verification here
            PrepareRequest message = context.GetPrepareRequestMessage(payload);
```

Aheuahueaea